pkg/virt-launcher/virtwrap/efi/efi.go

```diff
@@ -0,0 +1,118 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2021 Red Hat, Inc.
+ *
+ */
+
+package efi
+
+import (
+	"os"
+	"path/filepath"
+)
+
+const (
+	EFICode           = "OVMF_CODE.fd"
+	EFIVars           = "OVMF_VARS.fd"
+	EFICodeAARCH64    = "AAVMF_CODE.fd"
+	EFIVarsAARCH64    = "AAVMF_VARS.fd"
+	EFICodeSecureBoot = "OVMF_CODE.secboot.fd"
+	EFIVarsSecureBoot = "OVMF_VARS.secboot.fd"
+)
+
+type EFIEnvironment struct {
+	code           string
+	vars           string
+	codeSecureBoot string
+	varsSecureBoot string
+}
+
+func (e *EFIEnvironment) Bootable(secureBoot bool) bool {
+	if secureBoot {
+		return e.varsSecureBoot != "" && e.codeSecureBoot != ""
+	} else {
+		return e.vars != "" && e.code != ""
+	}
+}
+
+func (e *EFIEnvironment) EFICode(secureBoot bool) string {
+	if secureBoot {
+		return e.codeSecureBoot
+	} else {
+		return e.code
+	}
+}
+
+func (e *EFIEnvironment) EFIVars(secureBoot bool) string {
+	if secureBoot {
+		return e.varsSecureBoot
+	} else {
+		return e.vars
+	}
+}
+
+func DetectEFIEnvironment(arch, ovmfPath string) *EFIEnvironment {
+	if arch == "arm64" {
+		var codeArm64, varsArm64 string
+
+		_, err := os.Stat(filepath.Join(ovmfPath, EFICodeAARCH64))
+		if err == nil {
+			codeArm64 = EFICodeAARCH64
+		}
+		_, err = os.Stat(filepath.Join(ovmfPath, EFIVarsAARCH64))
+		if err == nil {
+			varsArm64 = EFIVarsAARCH64
+		}
+
+		return &EFIEnvironment{
+			code: filepath.Join(ovmfPath, codeArm64),
+			vars: filepath.Join(ovmfPath, varsArm64),
+		}
```

Comment:
If `ovmfPath/EFICodeAARCH64` and(/or) `ovmfPath/EFIVarsAARCH64` don't exist, `code` and(/or) `vars` will be set to `ovmfPath`, right?
Shouldn't we return nil, or at least set both to the empty string so that `Bootable()` fails?

Reply:
yes, you're right, made a mistake while refactoring, thanks for caching this!